Brucheion/Brucheion
# Use Go v1.16 embeds, use environment app keys

Closes #48.

### Functionality this PR provides:

* [x] Ditch pkger for [Go embedded files](https://golang.org/doc/go1.16#library-embed) (added with v1.16). This will hugely simplify setting up a working Brucheion development environment.
* [x] Use a `providers.json` file for storing authentication provider credentials outside of `config.json`. This file too is embedded into the Brucheion executable using Go 1.16 embeds.
* [x] Transfer credentials from CI environment secrets (GitHub Actions secrets) into said file.
* [x] Add credentials as secrets to GitHub Actions
* [x] Create a test CI build

### How to test:

* You will need the most recent Go v1.16 release, which should be available via most package managers (a `brew update` will be sufficient on macOS). Most changes will affect CI builds which we can test by creating a test alpha release.
* Create both development (`go build`) and production builds (`make release`) and test them individually.
